utils.go

```diff
@@ -146,3 +166,35 @@ func (s *Storage) getFileName(path string) string {
 		return path
 	}
 }
+
+func initCache(nc int64, mc int64, bi int64) (cache *Cache, err error) {
+	config := &Config{
+		NumCounters: nc,
+		MaxCost:     mc,
+		BufferItems: bi,
+	}
+
+	cache, err = NewCache(config)
+
+	if err != nil {
+		return nil, err
+	}
+	return cache, nil
+}
+
+func (s *Storage) setCache(path string, fileId string) (success bool) {
+	success = s.cache.SetWithTTL(path, fileId, cost, expireTime)
+	if success {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (s *Storage) getCache(path string) (string, bool) {
+	id, found := s.cache.Get(path)
+	if found {
+		return fmt.Sprintf("%v", id), true
```

Comment:
```suggestion
		return string(id), true
```

Maybe we can convert to string directly?

Reply:
It looks like we can't do conversion in this way. See this [https://play.golang.org/p/Vp64-ozhfdJ](https://play.golang.org/p/Vp64-ozhfdJ)

Reply:
> It looks like we can't do conversion in this way. See this https://play.golang.org/p/Vp64-ozhfdJ

Hi, I can't visit play.golang.org for "Unavailable For Legal Reasons", can you paste the code directly here?

Reply:
Code:
```
package main

import (
	"fmt"
)

func f() interface{} {
	return "str"
}
func main() {
	//fmt.Println(fmt.Sprintf("%v",f()))
	fmt.Println(string(f()))
}
```

Error:
```
cannot convert f() (type interface {}) to type string: need type assertion
```

Reply:
Ohhh, I got it.

We can use following code:

```go
return id.(string), true
```